From 594a7e9bce24aa38bf6c01501240de20968659c7 Mon Sep 17 00:00:00 2001
From: Hans-Kristian Arntzen <post@arntzen-software.no>
Date: Tue, 29 Mar 2022 15:03:35 +0200
Subject: [PATCH] dxil-spirv: report the failed DXIL opcode

Backport upstream diagnostic commit 594a7e9b.  The generic conversion error
does not identify which builder rejected the instruction, making a focused
compatibility backport impossible to select from device logs.
---
 .../dxil-spirv/opcodes/opcodes_dxil_builtins.cpp     | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/subprojects/dxil-spirv/opcodes/opcodes_dxil_builtins.cpp b/subprojects/dxil-spirv/opcodes/opcodes_dxil_builtins.cpp
index 74bf633a..6e97e6c3 100644
--- a/subprojects/dxil-spirv/opcodes/opcodes_dxil_builtins.cpp
+++ b/subprojects/dxil-spirv/opcodes/opcodes_dxil_builtins.cpp
@@ -356,7 +356,12 @@ bool emit_dxil_instruction(Converter::Impl &impl, const llvm::CallInst *instruct
 		return false;
 	}
 
-	return global_dispatcher.builder_lut[opcode](impl, instruction);
+	if (!global_dispatcher.builder_lut[opcode](impl, instruction))
+	{
+		LOGE("Failed DXIL opcode %u.\n", opcode);
+		return false;
+	}
+	return true;
 }
 
 static void update_raw_access_tracking_from_vector_type(Converter::Impl::AccessTracking &tracking,
-- 
2.43.0
